packages/flutter_tools/test/web.shard/hot_reload_web_test.dart

```diff
@@ -15,6 +15,7 @@ import 'test_data/hot_reload_index_html_samples.dart';
 
 void main() async {
   await _testProject(HotReloadProject()); // default
+  await _testProject(HotReloadProject(constApp: true), name: 'Default) (with `const MyApp()`)'); // runApp(const MyApp());
```

Comment:
@nshahan Ooh, it looks like this test harness will conditionally check for the following environment variables, and if provided, forward them to the underlying flutter tool it invokes: https://github.com/ditman/flutter-flutter/blob/web-test-hot-restart-const-app/packages/flutter_tools/test/integration.shard/test_utils.dart#L73

I'm guessing you want to run this with something like:

```
$ cd /path/to/framework/checkout
$ cd packages/flutter_tools
$ FLUTTER_LOCAL_ENGINE=host_debug_unopt FLUTTER_LOCAL_ENGINE_HOST=host_debug_unopt ../../bin/dart test test/web.shard/hot_reload_web_test.dart
```